zip-0315.rst

```diff
@@ -1,9 +1,620 @@
+
+Sent transactions
+'''''''''''''''''
+
+A transaction is sent if it was either:
+* created by the wallet, or
+* detected using the wallet's outgoing viewing keys, or
+* detected as having been sent from one of the wallet's watch-only addresses.
+
+Sent transactions SHOULD be reported with their number of confirmations,
+how long until they expire, and their balances as described in
+`Reporting of balances`_.
+
+
+Transaction creation
+--------------------
+
+Obtaining user consent for information leakage
+''''''''''''''''''''''''''''''''''''''''''''''
+
+Information leakage analysis
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The privacy provided by a Zcash transaction depends on the information leaked
+in the creation of that transaction and the process of it being broadcast for
+inclusion in the block chain.
+
+The requirements in this section are intended to minimize the leakage of such
+information where possible, and to ensure that the user is informed of any
+remaining information that would be leaked, and consents to such leakage.
+
+The list below describes the kinds of information that might be leaked. After
+a candidate transaction has been created and prior to it being revealed
+outside a trusted path to the user, the wallet user interface SHOULD obtain
+the user's consent for all of the leaked information.
+
+Assumption: There is always a transaction confirmation step for transactions
+that send funds out of the wallet.
+
+
+Kinds of information leakage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Transaction version (v4 or v5, as of NU5)
+
+SHOULD use v5 (unless you're spending Sprout funds).
+
+* Lock time (rarely used; may be a distinguisher if it is)
+
+SHOULD be zero.
+
+* Expiry height and anchor position
+
+See `Anchor selection`_ below.
+
+These give information about what block height the creator was synced to, and
+some policy information.
+
+* Transparent inputs and outputs
+
+See `Linkability of transactions or addresses`_.
+
+* Public value balances
+
+  * Together with the transparent inputs and outputs, these determine the fee
+    and the amount being transferred between pools.
+  * The fee is dependent on policy, but for a given policy we attempt to make
+    it only depend on other already-leaked metadata.
+
+SHOULD try to create fully shielded transactions where possible.
+
+Open question: Are we going to describe the policy that zcashd uses?
+
+
+* Numbers of JoinSplits, Spends, Outputs, and Actions
+
+  * This could correlate with other information under certain circumstances.
+    For example in a “dusting attack”, the adversary sends a victim lots of
+    small-valued notes (or notes with strategically chosen values),
+    increasing the probability that the victim’s transactions will have a
+    larger number of spends than other transactions (or an identifiable
+    number of spends). There are note management strategies that can mitigate
+    this, but they have not been implemented yet.
```

Comment:
see ZIP 317, which should make dusting attacks more expensive

Reply:
ZIP 317 will likely also alter wallet behavior with respect to how and when low-valued notes are included as inputs, and we may want to take that into account. Notes with less value than the marginal per-action fee will likely remain unspent unless it is possible to use them as padding. Pool-crossing transactions will not be able to take advantage of the padding allowance, because we expect all available action slots to be consumed by the economically relevant parts of the transaction. 

Wallets might also have policies that prevent TXOs from being spent if a substantial part of their value is lost to fees; we should make recommendations about that as well.

Reply:
I don't think that ZIP 317 affects what is said here other than in the specific note values that an adversary would choose.